internal/api/client.go

```diff
@@ -17,25 +20,50 @@ type ClientConfig struct {
 	AccessToken      string
 	OrganizationSlug string
 	ServerBaseUrl    string
+	DebugEnabled     bool
 }
 
 // authTransport is a middleware for the HTTP client.
 type authTransport struct {
-	accessToken string
+	accessToken  string
+	debugEnabled bool
 }
 
 // RoundTrip adds the Authorization header to all requests made by the HTTP client.
 func (t *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if t.debugEnabled {
+		fmt.Println("--- 🕵️ DEBUG")
+		fmt.Println("Request:")
```

Comment:
Consider using [`httputil.DumpRequestOut`](https://pkg.go.dev/net/http/httputil#DumpRequestOut), which will give you a more complete picture of the outgoing request.